src/scripts/main.js

```diff
@@ -1,3 +1,258 @@
 'use strict';
 
-// write code here
+const DESC = 'desc';
+const ASC = 'asc';
+const HEADERS = ['name', 'position', 'office', 'age', 'salary'];
+const OFFICES = [
+  'Tokyo',
+  'Singapore',
+  'London',
+  'New York',
+  'Edinburgh',
+  'San Francisco',
+];
+const DELAY_NOTIFICATION = 2000;
+
+function mapTableRowsToObjects(collection) {
+  return [...collection].map((item) => {
+    const data = item.querySelectorAll('td');
+
+    return HEADERS.reduce(
+      (prev, current, index) => ({
+        ...prev,
+        [current]: data[index].innerText,
+      }),
+      {},
+    );
+  });
+}
+
+function createRow(item) {
+  const row = document.createElement('tr');
+
+  for (const key in item) {
+    const rowData = document.createElement('td');
+
+    rowData.innerText = item[key];
+    row.appendChild(rowData);
+  }
+
+  return row;
+}
+
+function rebuildList(data) {
+  const tableBody = document.querySelector('tbody');
+
+  tableBody.innerHTML = '';
+
+  for (const item of data) {
+    const row = createRow(item);
+
+    tableBody.appendChild(row);
+  }
+}
+
+function sortItems(items, targetValue, order) {
+  items.sort((a, b) => {
```

Comment:
The `sortItems` function is doing too many things at once, which makes it hard to read and maintain. I suggest splitting it up into several smaller functions, each of which sorts the items based on a specific attribute (e.g., `sortBySalary`, `sortByAge`, `sortByName`). This will make your code more modular and easier to understand.